### PR TITLE
Require only s1 or b1

### PR DESCRIPTION
**Pull request by EricKutschera**
* The extra plots from s2 and b2 are not necessary
* This is useful for plotting a single SAM or BAM file

Requested here: https://github.com/Xinglab/rmats-turbo/issues/28#issuecomment-664538124

### CODE REVIEW

File: src/rmats2sashimiplot/rmats2sashimiplot.py [hunk 1]
```diff
@@ -6,32 +6,41 @@
...
-    for bam in options.b2.split(","):
-        if os.path.isfile(bam + '.bai'):
-            print("'{0}' is indexed already: '{0}.bai'".format(bam))
-        else:
-            print("Indexing '{}'.".format(bam))
-            os.system("samtools index " + bam)
-    return
+
+
+def prepare_bams(options):
```
Comment (ywang1993) [inline src/rmats2sashimiplot/rmats2sashimiplot.py:31]: For cases that users only provide s2/b2 without s1/b1. The indent of the if statement '`if options.s2:`' should be decreased. Also, there should be another if statement for the  'options.b1'.

File: src/rmats2sashimiplot/rmats2sashimiplot.py [hunk 2]
```diff
@@ -63,37 +72,39 @@ def checkout(parser, options):
     """
     # bam files and sam files are alternative, the same for the case of events_file and coordinate
     # events_file should be provided together with event_type
-    if (options.s1 is None and options.b1 is None) or (options.s2 is None and options.b2 is None):
-        parser.error("Not enough arguments! Please provide sam or bam files.")
+    if options.s1 is None and options.b1 is None:
```
Comment (ywang1993) [inline src/rmats2sashimiplot/rmats2sashimiplot.py:75]: `if (options.s1 is None and options.b1 is None) and (options.s2 is None and options.b2 is None):`

File: src/rmats2sashimiplot/rmats2sashimiplot.py [hunk 3]
```diff
@@ -63,37 +72,39 @@ def checkout(parser, options):
...
+    if (((options.s1 is not None or options.s2 is not None)
+         and (options.b1 is not None or options.b2 is not None))):
+        parser.error("Specify either sam files or bam files not both")
     if (options.events_file is None or options.event_type is None) and options.coordinate is None:
         parser.error("Not enough arguments! Please provide "
                      "1) coordinates with gff3 files. or "
                      "2) events files together with events type.")
 
-    if options.s1 is not None and options.s2 is not None:  # with sam file
+    if options.s1 is not None:  # with sam file
```
Comment (ywang1993) [inline src/rmats2sashimiplot/rmats2sashimiplot.py:85]: `if options.s1 is not None or options.s2 is not None:`

File: src/rmats2sashimiplot/rmats2sashimiplot.py [hunk 4]
```diff
@@ -63,37 +72,39 @@ def checkout(parser, options):
...
-        parser.error("Not enough arguments! Please provide sam or bam files.")
+    if options.s1 is None and options.b1 is None:
+        parser.error("Not enough arguments! Please provide either --s1 or --b1")
+    if (((options.s1 is not None or options.s2 is not None)
+         and (options.b1 is not None or options.b2 is not None))):
+        parser.error("Specify either sam files or bam files not both")
     if (options.events_file is None or options.event_type is None) and options.coordinate is None:
         parser.error("Not enough arguments! Please provide "
                      "1) coordinates with gff3 files. or "
                      "2) events files together with events type.")
```
Comment (ywang1993) [inline src/rmats2sashimiplot/rmats2sashimiplot.py:84]: `if options.s2 is not None:`
`elif options.b2 is not None`
decrease indentation

**Review by ywang1993 — COMMENTED** [review]
Changes need to be make considering that users might only provide s2/b2 without s1/b1.

**Comment by EricKutschera** [comment]
The latest commit updates the `if` statements as suggested and also updates other places in the code that were expecting to always have the first sample

**Review by ywang1993 — APPROVED** [review]

**Pull request merged**